controllers/reconcile_cli_test.go

```diff
@@ -94,46 +91,114 @@ var _ = Describe("Reconcile CLI", func() {
-					_, err := time.Parse(time.RFC3339, annotations["rabbitmq.com/queueRebalanceNeededAt"])
+					Expect(fakeExecutor.ExecutedCommands()).NotTo(ContainElement(command{"bash", "-c", "rabbitmqctl enable_feature_flag all"}))
+					_, err := time.Parse(time.RFC3339, rmq.Annotations["rabbitmq.com/queueRebalanceNeededAt"])
 					Expect(err).NotTo(HaveOccurred(), "Annotation rabbitmq.com/queueRebalanceNeededAt was not a valid RFC3339 timestamp")
 				})
 
 				By("removing the annotation once all Pods are up, and triggering the queue rebalance", func() {
+					// setup transition to all pods ready
 					sts.Status.ReadyReplicas = 3
 					Expect(client.Status().Update(ctx, sts)).To(Succeed())
-					Eventually(func() map[string]string {
-						rmq := &rabbitmqv1beta1.RabbitmqCluster{}
-						err := client.Get(ctx, types.NamespacedName{Name: cluster.Name, Namespace: cluster.Namespace}, rmq)
-						Expect(err).To(BeNil())
-						return rmq.ObjectMeta.Annotations
-					}, 5).ShouldNot(HaveKey("rabbitmq.com/queueRebalanceNeededAt"))
+
+					// by not having the annotation
+					rmq := &rabbitmqv1beta1.RabbitmqCluster{}
+					rmq.Name = "rabbitmq-three"
+					rmq.Namespace = defaultNamespace
+					Eventually(k.Object(rmq)).Within(time.Second * 5).WithPolling(time.Second).ShouldNot(HaveField("ObjectMeta.Annotations", HaveKey("rabbitmq.com/queueRebalanceNeededAt")))
+
+					// by executing the commands
 					Expect(fakeExecutor.ExecutedCommands()).To(ContainElement(command{"sh", "-c", "rabbitmq-queues rebalance all"}))
+					Expect(fakeExecutor.ExecutedCommands()).To(ContainElement(command{"bash", "-c", "rabbitmqctl enable_feature_flag all"}))
+				})
+			})
+		})
+
+		Describe("autoEnableAllFeatureFlags", func() {
+			var (
+				sts *appsv1.StatefulSet
+				k   komega.Komega
+			)
+
+			BeforeEach(func() {
+				k = komega.New(client)
+				sts = statefulSet(ctx, cluster)
+				sts.Status = appsv1.StatefulSetStatus{
+					Replicas:          3,
+					ReadyReplicas:     3,
+					CurrentReplicas:   3,
+					UpdatedReplicas:   3,
+					CurrentRevision:   "the last one",
+					UpdateRevision:    "the last one",
+					AvailableReplicas: 3,
+				}
+				sts.Annotations = make(map[string]string)
+				Expect(client.Status().Update(ctx, sts)).To(Succeed())
+			})
+
+			When("disabled", func() {
+				It("doesn't call enable_feature_flag CLI", func() {
+					// BeforeEach updated the STS from zero replicas ready to all ready
+					// Initial rabbitmqcluster create command has reconciles pending to have
+					// all replicas ready to execute commands. We have to reset the "registry"
+					// of commands.
+					fakeExecutor.ResetExecutedCommands()
+
+					Eventually(k.Update(cluster, func() {
+						if cluster != nil {
+							cluster.Spec.AutoEnableAllFeatureFlags = false
+						}
+					})).Should(Succeed())
+					Eventually(fakeExecutor.ExecutedCommands).Within(time.Second * 5).WithPolling(time.Second).
```

Comment:
I think we should assert with `Consistently` instead of `Eventually`, because you may assert too soon, and eventually will pass, even if the reconcile loop in the background runs the command later.

If I understand the intention of this test, correctly, we want to ensure that the command is not run; you can't guarantee that without `Consistently`.